AirCasting/CoreData/CoreDataHook.swift

```diff
@@ -0,0 +1,40 @@
+// Created by Lunar on 17/05/2021.
+//
+
+import Foundation
+import CoreData
+
+class CoreDataHook: NSObject, ObservableObject {
+    
+    @Published var sessions: [Session] = []
+    var context: NSManagedObjectContext!
+    
+    var fetchRequest: NSFetchRequest<Session>! {
```

Comment:
seams like You are just editing the predicate. So maybe make it private lazy and expose the setup method that accepts the `SelectedSection` ? Also the NSFetchedResultsController can be a lazy var. just the performFetch() needs to be called after the predicate change.